### PR TITLE
Make generated jobs generate to its own config file

### PR DESCRIPTION
**Pull request by krzyzacy**
Gonna need to cleanup old configs & probably fix a bunch of pylint
/hold
/assign @BenTheElder @yguo0905

### CODE REVIEW

File: experiment/generate_tests.py [hunk 1]
```diff
@@ -281,7 +270,7 @@ def for_each_job(output_dir, job_name, job, yaml_config):
         generator = E2ENodeTest(job_name, job, yaml_config)
     else:
         raise ValueError('Unexpected job type ', job_type)
-    envs, job_config, prow_config = generator.generate()
+    job_config, prow_config = generator.generate()
```
Comment (BenTheElder) [inline experiment/generate_tests.py]: this doesn't look like update bazel? /shrug guess the commits don't matter too much

Comment (krzyzacy) [reply experiment/generate_tests.py]: I probably amended into that commit :-)

File: config/jobs/kubernetes/generated/generated.yaml [hunk 2]
```diff
@@ -0,0 +1,6395 @@
+periodics:
+- tags:
+  - generated   # AUTO-GENERATED by experiment/generate_tests.py - DO NOT EDIT!
```
Comment (BenTheElder) [inline config/jobs/kubernetes/generated/generated.yaml]: are these all node tests? if so why not move this generator machinery into kubernetes/sig-node/ and make yguo0905 an approver there ...?

Comment (krzyzacy) [reply config/jobs/kubernetes/generated/generated.yaml]: no not all of them... node + image validation + some release validation jobs... We can add some selector to generate multiple files in the future.

Comment (BenTheElder) [reply config/jobs/kubernetes/generated/generated.yaml]: ok

File: experiment/generate_tests.py [hunk 3]
```diff
@@ -274,13 +233,6 @@ def generate(self):
         k8s_version = self.k8s_versions[fields[5][3:]]
         test_suite = self.test_suites[fields[6]]
 
-        # Generates envs.
-        envs = []
-        envs.extend(get_envs(self.job_name, 'common', self.common))
```
Comment (yguo0905) [inline experiment/generate_tests.py]: We can remove this part because no configs use envs?

**Comment by yguo0905** [comment]
This is super cool! You've simplified the generation tool and configs a lot!

File: experiment/generate_tests.py (hunk 3 above)
Comment (krzyzacy) [reply experiment/generate_tests.py]: I believe those envs are written out as the `--env-file=job.env`? I've checked nothing writes out env files anymore.

File: config/jobs/kubernetes/generated/generated.yaml [hunk 4]
```diff
@@ -0,0 +1,6395 @@
...
+    preset-k8s-ssh: 'true'
+  name: ci-kubernetes-e2e-gke-ubuntu1-k8sstable2-autoscaling
+  spec:
+    containers:
+    - args:
+      - --bare
+      - --timeout=620
+      - --scenario=kubernetes_e2e
+      - --
+      - --cluster=test-33e1c302
```
Comment (krzyzacy) [inline config/jobs/kubernetes/generated/generated.yaml]: the individual envs are still being loaded, which is WAI. cc @yguo0905

**Comment by krzyzacy** [comment]
I'll merge this Monday

File: experiment/generate_tests.py (hunk 3 above)
Comment (yguo0905) [reply experiment/generate_tests.py]: You are right. I was thinking about the `--env` flags, but I think they are handled as `args`.

**Comment by yguo0905** [comment]
/lgtm
/hold
Thanks!

**Comment by krzyzacy** [comment]
github diff is soooooo useless :-(

**Comment by krzyzacy** [comment]
I think I rebased properly... I think... let's see...

**Comment by krzyzacy** [comment]
checked with https://www.diffchecker.com/ looks correct, but cannot create a free saved version :-)

**Comment by krzyzacy** [comment]
wanna re-stamp?

File: README.md [hunk 5]
```diff
@@ -97,7 +97,7 @@ Our build job will generate a ci/(channel-name) file pointer in gcs.
 After you update [test_config.yaml](experiment/test_config.yaml), please run
 
 ```
-bazel run //experiment:generate_tests -- --yaml-config-path=experiment/test_config.yaml --json-config-path=jobs/config.json  --prow-config-path=prow/config.yaml && bazel run //jobs:config_sort
+bazel run //experiment:generate_tests -- --yaml-config-path=experiment/test_config.yaml --json-config-path=jobs/config.json  --prow-config-path=prow/config.yaml
```
Comment (BenTheElder) [inline README.md]: this example is now stale

Comment (krzyzacy) [reply README.md]: ack

Comment (krzyzacy) [reply README.md]: looks like it's fixed in your PR, thanks :-)

**Pull request merged**